polaris-react/src/components/Button/Button.scss

```diff
@@ -758,49 +730,46 @@
     min-height: var(--p-font-line-height-2);
     min-width: var(--p-font-line-height-2);
 
-    svg {
-      fill: var(--p-color-icon-subdued);
-    }
+    // stylelint-disable-next-line -- Button custom properties
+    --pc-button-icon-color: var(--p-color-icon-subdued);
 
-    &:hover svg {
-      fill: var(--p-color-icon-hover);
+    &:hover {
+      // stylelint-disable-next-line -- Button custom properties
+      --pc-button-icon-color: var(--p-color-icon-hover);
     }
 
-    &:active svg,
-    &.pressed svg {
-      fill: var(--p-color-icon-active);
+    &:active,
+    &.pressed {
+      // stylelint-disable-next-line -- Button custom properties
+      --pc-button-icon-color: var(--p-color-icon-active);
     }
 
-    &:focus svg {
-      fill: var(--p-color-icon-hover);
+    &:focus {
+      // stylelint-disable-next-line -- Button custom properties
+      --pc-button-icon-color: var(--p-color-icon-hover);
     }
 
-    &.disabled svg {
-      fill: var(--p-color-icon-disabled);
+    &.disabled {
+      // stylelint-disable-next-line -- Button custom properties
+      --pc-button-icon-color: var(--p-color-icon-disabled);
     }
   }
 
   &.critical.iconOnly {
-    svg {
-      fill: var(--p-color-icon-critical-strong-experimental);
-    }
-
-    &:hover svg {
-      fill: var(--p-color-icon-critical-strong-hover-experimental);
-    }
-
-    &:focus:not(:active):not(.pressed):not(.disabled) svg {
-      fill: var(--p-color-icon-critical-strong-experimental);
-    }
+    // stylelint-disable-next-line -- Button custom properties
+    --pc-button-icon-color: var(--p-color-icon-critical-strong-experimental);
+  }
 
-    &:active svg,
-    &.pressed svg {
-      fill: var(--p-color-icon-critical-strong-active-experimental);
-    }
+  &:hover {
+    // stylelint-disable-next-line -- Button custom properties
+    --pc-button-icon-color: var(
+      --p-color-icon-critical-strong-hover-experimental
+    );
+  }
 
-    &.disabled svg {
-      fill: var(--p-color-icon-disabled);
-    }
```

Comment:
The two declarations were duplicates once the unneeded `svg` tag is removed from it, so the custom property is set at the top of the equivalent class' declarations on 606 and 664.

Reply:
(ignore)